core/src/main/scala/org/apache/spark/ContextAwareIterator.scala

```diff
@@ -28,10 +29,12 @@ import org.apache.spark.TaskContext
  * which crashes the executor.
  * Thus, we should use [[ContextAwareIterator]] to stop consuming after the task ends.
  */
-class ContextAwareIterator[IN](iter: Iterator[IN], context: TaskContext) extends Iterator[IN] {
+@DeveloperApi
+class ContextAwareIterator[+T](val context: TaskContext, val delegate: Iterator[T])
```

Comment:
Nice! Thank you for revising this consistently with `InterruptibleIterator`. Now, it looks much better.